src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -584,6 +584,16 @@ bool Connection::websocket_closed_handler(bool was_clean, WebSocketError error_c
             break;
         }
         case WebSocketError::websocket_fatal_error: {
+            // If the connection fails/times out and the server has not been contacted yet, refresh the location
+            // to make sure the websocket URL is correct
+            if (!m_server_endpoint.is_verified) {
+                SessionErrorInfo error_info(
+                    {ErrorCodes::SyncConnectFailed, util::format("Failed to connect to sync: %1", msg)},
```

Comment:
When do we typically get this error? I think `ErrorCodes::ConnectionClosed` is could be the more accurate error code but `RefreshLocation` is definitely the right action to take.

Reply:
It is used here, when the websocket handshake response isn't one of the expected status codes (e.g. 404).
https://github.com/realm/realm-core/blob/mwb/location-failure/src/realm/sync/network/default_socket.cpp#L122

Should I just update this to always refresh the location, regardless of whether or not the sync route has been verified?